frontend/src/individual_add.tsx

```diff
@@ -158,6 +159,51 @@ export function IndividualAdd({
     };
   });
 
+  // creates a breeding event, updates it with birth information and calls the createIndividual function
+  const prepareIndividual = async () => {
+    let breedingString = "";
+    if (individual.birth_date) {
+      let breedingDate: Date | number = new Date(individual.birth_date);
+      breedingDate.setDate(breedingDate.getDate() - 30);
+      breedingString = breedingDate.toLocaleDateString(locale);
+
+      let breeding = {
+        mother: individual.mother?.number,
+        father: individual.father?.number,
+        date: breedingString,
+      };
+
+      let birth = {
+        date: individual.birth_date,
+        litter: individual.litter,
+        id: null,
+      };
+
+      const breedingEvent = await post("/api/breeding", breeding);
+
```

Comment:
Error handling? 

And we should probably check if this breeding exist before trying to create it (as mentioned, I don't know if the API is there, but we can add if needed).

Reply:
Right. My thought was to have this one, pretty generic error handling for both requests:
```
 } else {
        userMessage(
          // register_birth in data_access.py requires a litter size.
          // Thus, the field must be mandatory in the form. Okay for föreningen?
          "Något gick fel. Kolla så att du har lagt till mor, far, kullstorlek och födelsedatum.",
          "error"
        );
      }
```
It will be triggered if something in one of the requests goes wrong, but maybe it's not so beautiful. I can add more specific error handling to the first request to prevent sending a second request that is meant to fail from the very beginning. 

Also, the backend checks if there already is a breeding, it looks like this in `register_breeding` in `data_access.py`: 
```
exists = (
        Breeding.select()
        .where(Breeding.father == father)
        .where(Breeding.mother == mother)
        .where(Breeding.breed_date == breed_date)
        .count()
    )
    if exists:
        return {"status": "error", "message": "Breeding already registered"}
```
But I would need the endpoint to send the breeding's id even in this case. For now, it's only send in case a new breeding is created. So that's something that needs to be added, and I don't think I can do that myself.

Reply:
I don't see a problem with a generic error message, but if I understand things correctly, that relies on doing the second request and failing - is that incorrect?

(I would possibly expect to get some other error because of `breeding_id` not being present in `breedingEvent`, but I guess that doesn't happen.)

Reply:
Regarding backend; if it simplifies I guess we can change the add breeding api endpoint to allow existing breeding to just return that. The alternative would be to get the breeding for the herd and check if it exists already.

Reply:
So now we check if the breeding exists by getting all the breedings for the current herd. Only there is no breeding a request is sent to create one.